oauth2client/client.py

```diff
@@ -272,7 +272,7 @@ def new_from_json(cls, s):
       An instance of the subclass of Credentials that was serialized with
       to_json().
     """
-    if six.PY3 and isinstance(s, bytes):
```

Comment:
json input should always be an unicode string, this just makes sure we always utf-8 decode the input before passing it to json.loads instead of having json.loads (on python 2) deal (and possibly fail) with the encoding.

I'd say the code should (mostly) just handle unicode strings, but for compatibility it's easier to allow utf-8 encoded bytestring inputs.  But we should probably implement a utf-8-bytes-accepting json decoding function in util.py and use it everywhere instead of duplicating the type check/decode/json load code everywhere.